services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+ */
+
+package software.amazon.awssdk.services.sqs.batchmanager;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeName;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration values for the BatchManager Implementation.  All values are optional, and the default values will be used if they
+ * are not specified.
+ */
+@SdkPublicApi
+public final class BatchOverrideConfiguration implements ToCopyableBuilder<BatchOverrideConfiguration.Builder,
+    BatchOverrideConfiguration> {
+
+    private final Integer maxBatchItems;
+    private final Integer maxBatchKeys;
+    private final Integer maxBufferSize;
+    private final Duration maxBatchOpenDuration;
+    private final Duration visibilityTimeout;
+    private final Duration longPollWaitTimeout;
+    private final Duration minReceiveWaitTime;
+    private final Integer maxDoneReceiveBatches;
+    private final List<MessageSystemAttributeName> messageSystemAttributeNames;
+    private final List<String> receiveMessageAttributeNames;
+    private final Boolean adaptivePrefetching;
+    private final Integer maxInflightReceiveBatches;
+
+    public BatchOverrideConfiguration(Builder builder) {
+        this.maxBatchItems = Validate.isPositiveOrNull(builder.maxBatchItems, "maxBatchItems");
+        this.maxBatchKeys = Validate.isPositiveOrNull(builder.maxBatchKeys, "maxBatchKeys");
+        this.maxBufferSize = Validate.isPositiveOrNull(builder.maxBufferSize, "maxBufferSize");
+        this.maxBatchOpenDuration = Validate.isPositiveOrNull(builder.maxBatchOpenDuration, "maxBatchOpenDuration");
+        this.visibilityTimeout = Validate.isPositiveOrNull(builder.visibilityTimeout, "visibilityTimeout");
+        this.longPollWaitTimeout = Validate.isPositiveOrNull(builder.longPollWaitTimeout, "longPollWaitTimeout");
+        this.minReceiveWaitTime = Validate.isPositiveOrNull(builder.minReceiveWaitTime, "minReceiveWaitTime");
+        this.messageSystemAttributeNames = builder.messageSystemAttributeNames;
+        this.receiveMessageAttributeNames = builder.receiveMessageAttributeNames;
+        this.adaptivePrefetching = builder.adaptivePrefetching;
+        this.maxInflightReceiveBatches = builder.maxInflightReceiveBatches;
+        this.maxDoneReceiveBatches = builder.maxDoneReceiveBatches;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * @return the optional maximum number of messages that are batched together in a single request.
+     */
+    public Integer maxBatchItems() {
+        return maxBatchItems;
+    }
```

Comment:
Is this for both request and response?

Reply:
This is used only for SendRequest like SendMessage/ChangeVisibilty/Delete and not for RecieveMessage